src/thunderbots/software/util/visualizer_messenger/visualizer_messenger.cpp

```diff
@@ -29,6 +29,19 @@ namespace Util
 
     void VisualizerMessenger::publishAndClearLayers()
     {
+        // Limit rate of the message publishing
+        // Get the time right now
+        const time_point now  = std::chrono::system_clock::now();
+        const auto elapsed_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(
+                                    now - time_last_published)
+                                    .count();
+
+        // Do not do anything if the time passed hasn't been
+        // long enough
+        if (elapsed_ns < DESIRED_PERIOD_NS)
```

Comment:
We generally use seconds or milliseconds, would either of those be reasonable to use here? Or is there a specific reason we chose nanoseconds?

Reply:
I think milliseconds would be fine. nanoseconds seems like overkill

Reply:
I think the reason I used nanoseconds is because if I had used milliseconds, the difference of `now - time_last_published` gets rounded to 0, because this function is being called at >> 1000 times a second. 

Any workarounds? 

Also isn't using `chrono::nanoseconds` just a scaling ratio of the system clock ticks? So it would have the same performance?

Reply:
Ah, I was thinking we were getting floating point values. Could we just scale the nanoseconds to milliseconds? It's mostly about being consistent with the units we use in the rest of the codebase.

Reply:
Ah didn't realize rounding was an issue. I'd be fine leaving it as it is just with a comment explaining why we use nanoseconds (basically what you said above)

Reply:
I'm fine with mats suggestion as well, your call @FSXAC